packages/flame/lib/src/widgets/animation_widget.dart

```diff
@@ -2,169 +2,62 @@ import 'dart:math';
+        return CustomPaint(
+          painter: SpritePainter(
+            controller.animation.getSprite(),
+            anchor,
+          ),
         );
       },
-      errorBuilder: errorBuilder,
-      loadingBuilder: loadingBuilder,
     );
   }
 }
 
-/// A [StatefulWidget] that render a [SpriteAnimation].
-class _SpriteAnimationWidget extends StatefulWidget {
-  /// The [SpriteAnimation] to be rendered
-  final SpriteAnimation animation;
-
-  /// The positioning [Anchor]
-  final Anchor anchor;
-
-  /// Should the [animation] be playing or not
-  final bool playing;
-
-  const _SpriteAnimationWidget({
+class SpriteAnimationController extends AnimationController {
+  SpriteAnimationController({
+    required TickerProvider vsync,
     required this.animation,
-    this.playing = true,
-    this.anchor = Anchor.topLeft,
-  });
-
-  @override
-  State createState() => _SpriteAnimationWidgetState();
-}
-
-class _SpriteAnimationWidgetState extends State<_SpriteAnimationWidget>
-    with SingleTickerProviderStateMixin {
-  AnimationController? _controller;
-  double? _lastUpdated;
-
-  @override
-  void initState() {
-    super.initState();
-    widget.animation.onComplete = _pauseAnimation;
-    _setupController();
-    if (widget.playing) {
-      _initAnimation();
-    }
-  }
-
-  @override
-  void didUpdateWidget(_SpriteAnimationWidget oldWidget) {
-    super.didUpdateWidget(oldWidget);
-
-    if (oldWidget.animation != widget.animation) {
-      oldWidget.animation.onComplete = null;
-      _setupController();
-    }
-
-    if (widget.playing) {
-      _initAnimation();
-    } else {
-      _pauseAnimation();
-    }
-  }
-
-  void _initAnimation() {
-    setState(() {
-      widget.animation.reset();
-      _lastUpdated = DateTime.now().millisecond.toDouble();
-      _controller?.repeat(
-        // Approximately 60 fps
-        period: const Duration(milliseconds: 16),
-      );
-    });
+  }) : super(vsync: vsync) {
+    duration = Duration(seconds: animation.totalDuration().ceil());
   }
 
-  void _setupController() {
-    _controller?.dispose();
-
-    _controller = AnimationController(vsync: this)
-      ..addListener(() {
-        final now = DateTime.now().millisecond.toDouble();
-
-        final dt = max(0, (now - (_lastUpdated ?? 0)) / 1000).toDouble();
-        widget.animation.update(dt);
-
-        setState(() => _lastUpdated = now);
-      });
-  }
+  final SpriteAnimation animation;
 
-  void _pauseAnimation() {
-    setState(() => _controller?.stop());
-  }
+  double? _lastUpdated;
 
   @override
-  void dispose() {
-    _controller?.dispose();
-    super.dispose();
-  }
+  void notifyListeners() {
+    super.notifyListeners();
 
-  @override
-  Widget build(BuildContext ctx) {
-    return Container(
-      child: CustomPaint(
-        painter: SpritePainter(
-          widget.animation.getSprite(),
-          widget.anchor,
-        ),
-      ),
-    );
+    final now = DateTime.now().millisecond.toDouble();
```

Comment:
`DateTime.now().millisecond` returns the "millisecond" part of the datetime, a number between 0 and 999. Which means there will be animation jank every second.

Reply:
I copied it from the previous logic, so I think this issue was still in the previous code. I will look into this.